c/tests/integration/driver_test.c

```diff
@@ -0,0 +1,443 @@
+        if (check_error()) goto cleanup;
+        else concept_map_iterator_drop(insertResult);
+
+        ConceptMapIterator* it = query_match(transaction, "match $n isa name;", opts);
+        ConceptMap* conceptMap;
+        bool foundJohn = false;
+        while (0 != (conceptMap = concept_map_iterator_next(it)) && !check_error()) {
+            Concept* concept = concept_map_get(conceptMap, "n");
+            Concept* asValue = attribute_get_value(concept);
+            char* attr = value_get_string(asValue);
+            foundJohn = foundJohn || (0 == strcmp(attr, "John"));
+            free(attr);
+            concept_drop(asValue);
+            concept_drop(concept);
+            concept_map_drop(conceptMap);
+        }
+        concept_map_iterator_drop(it);
+
+        transaction_commit(transaction);
+        transaction = 0;
+
+        if (!foundJohn) {
+            fprintf(stderr, "Did not find inserted name \'John\' in query result.\n");
+            goto cleanup;
+        }
+    }
+
+    completed = true;
+
+cleanup:
+    if (check_error()) print_error(__FILE__, __LINE__);
+    transaction_drop(transaction);
+    session_drop(session);
+    options_drop(opts);
+
+    delete_database_if_exists(dbMgr, dbName);
+    if (check_error()) print_error(__FILE__, __LINE__);
+    database_manager_drop(dbMgr);
+    connection_close(conn);
+    return completed;
+}
+
+bool test_concept_api_schema() {
+    const char dbName[] = "test_concept_api";
+
+    Connection* conn = 0;
+    DatabaseManager* dbMgr = 0;
+    Session* session = 0;
+    Transaction* transaction = 0;
+    Options* opts = 0;
+
+    bool completed = false;
+
+    conn = connection_open_core(TYPEDB_CORE_ADDRESS);
+    if (check_error()) goto cleanup;
+
+    dbMgr = database_manager_new(conn);
+    if (check_error()) goto cleanup;
+
+    delete_database_if_exists(dbMgr, dbName);
+    if (check_error()) goto cleanup;
+
+    databases_create(dbMgr, dbName);
+    if (check_error()) goto cleanup;
+
+    opts = options_new();
+    if (check_error()) goto cleanup;
+
+    // test schema api
+    {
+        session = session_new(dbMgr, dbName, Schema, opts);
+        if (check_error()) goto cleanup;
+
+        transaction = transaction_new(session, Write, opts);
+        if (check_error()) goto cleanup;
+        {
+            Concept* definedNameType = concepts_put_attribute_type(transaction, "name", String);
+            if (check_error()) goto cleanup;
+            else concept_drop(definedNameType);
+        }
+
+        {
+            ConceptIterator* it = 0;
+            Concept* nameType = 0;
+            Concept* rootAttributeType = 0;
+            bool foundName = false;
+            if (
+                0 != (nameType = concepts_get_attribute_type(transaction, "name")) &&
+                0 != (rootAttributeType = concepts_get_attribute_type(transaction, "attribute")) &&
+                0 != (it = attribute_type_get_subtypes(transaction, rootAttributeType, Transitive))) {
```

Comment:
Seeing this made me realize that I actually prefer implicit null pointer checks to the explicit `!= NULL`. I don't know if it's exposure or a legitimate preference though, I'm still leaning towards keeping the explicit checks.